apple/testing/default_runner/simulator_creator.py

```diff
@@ -34,17 +34,48 @@ def _boot_simulator(simulator_id: str) -> None:
         output = _simctl(["bootstatus", simulator_id, "-b"])
         print(output, file=sys.stderr)
     except subprocess.CalledProcessError as e:
+        exit_code = e.returncode
+
+        # When reusing simulators we may encounter the error:
+        # 'Unable to boot device in current state: Booted'.
+        #
+        # This is because the simulator is already booted, and we can ignore it
+        # if we check and the simulator is in fact booted.
+        if exit_code == 149:
+            devices = json.loads(
+                _simctl(["list", "devices", "-j", simulator_id]),
```

Comment:
Should this call to `simctl` be wrapped in a `try/except` block as well?

Reply:
If this one fails I don't want to catch it. The `except` is to try to recover from specific error codes during booting, but this call is just getting the list of devices.